Project Program/TOC Project/src/com/car/Car.java

```diff
@@ -6,146 +6,265 @@
+			System.out.println("1. Accelerate");
+			System.out.println("2. Press Brake");
+			System.out.println("3. Hold Brake");
+			break;
+		case "IN-FORWARD-MOTION":
+			System.out.println("1. Accelerate");
+			System.out.println("2. Press Brake");
+			System.out.println("3. Activate Cruise Control");
+			break;
+		case "CRUISECONTROL":
+			System.out.println("1. Accelerate");
+			System.out.println("2. Press Brake");
+			break;
 		}
-		recognizer.stopRecognition();
-		
-		
-		LiveSpeechRecognizer recog = new LiveSpeechRecognizer(config);
-		// Start recognition process pruning previously cached data.
-		recog.startRecognition(true);
-		SpeechResult res = recognizer.getResult();
-		System.out.println("\n\n here");
-		System.out.println("Speech : ");//+command);
-		// Pause recognition process. It can be resumed then with startRecognition(false). */
-		//recognizer.stopRecognition();
-		
+			
 	}
-		
-	
 	
 	//Regular typing input
 	public void signal() {
 		try {
 			
-			Scanner in = new Scanner(System.in);
-		System.out.println("What signal are you giving? ");
-		signal = in.next();
-		
-		//HERE CAN USE IF STATEMENTS TO SEE WHERE THE CODE GOES FROM OFF STATE 
-		//TO MOVE THE CAR.
-		
-		//Gives a if statement to check what state it is in and if in relevant state and
-		// Is going to another relevant state then the speed changes.
-		if(state.toLowerCase().equals("stationary") && signal.toLowerCase().equals("drive")) {
-			
-			//IF it is correct, then the next state is given and the increase speed method then called.
-			increase_speed();
-			inForward_Motion();
-			
+		Scanner in = new Scanner(System.in);
+		System.out.println("Welcome to UTUVCS (Designed by RSC <3)");
+			while(true) {
+				//Displays current state of the vehicle
+				printCurrentState();
+				//Display options based on state of the vehicle
+				dispOptions();
+				System.out.println("What signal are you giving?(Enter the  first word in the chosen command) ");
+				signal = in.next();
+				//HERE CAN USE IF STATEMENTS TO SEE WHERE THE CODE GOES FROM OFF STATE 
+				//TO MOVE THE CAR.
+				
+				//We are using a case to check what state it is in and if in relevant state and
+				// Is going to another relevant state then the speed changes.
+				
+				//Switch statement to handle state transition
+			StateChanger(signal);
 		}
 		
-		if(state.toLowerCase().equals("stationary") && signal.toLowerCase().equals("reverse")) {
-			
-			//IF it is correct, then the next state is given and the increase speed method then called.
-			increase_speed();
-			inReverse_Motion();
-			
-		}
 		
 		}catch(NullPointerException e) {
 			System.out.println("Null pointer encountered at Signal method.");
 		}
 	}
 	
 	
-	public void inForward_Motion() {
-		
-		System.out.println("Gear Changed to drive + \n");
-		System.out.println("In-forward motion :) \n");
-		
+public  void StateChanger(String signal) //based on the inputted signal, it reads the car's state and acts accordingly

```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Reply:
Thank you for the review.

Reply:
Happy to help.